lib/rubycritic/commands/compare.rb

```diff
@@ -0,0 +1,203 @@
+        critic = critique('feature_branch_hash')
+        Config.feature_branch_score = critic.score
+        Config.root = feature_root_directory
+        Config.feature_branch_flag = Config.no_browser = true
+        report(critic)
+        Config.feature_branch_flag = false
+        switch_branch(Config.base_branch)
+      end
+
+      def feature_branch_analysis
+        Config.no_browser = false
+        defected_modules = @analysed_modules.where(degraded_files)
+        analysed_modules = AnalysedModulesCollection.new(defected_modules.map(&:path), defected_modules)
+        Config.root = build_directory
+        Config.set_location = Config.build_flag = true
+        @code_index_file_location = Reporter.generate_report(analysed_modules)
+        score_difference
+      end
+
+      def score_difference
+        Config.difference_score =
+          if Config.base_branch_score > Config.feature_branch_score
+            (Config.base_branch_score - Config.feature_branch_score).round(2)
+          else
+            (Config.feature_branch_score - Config.base_branch_score).round(2)
+          end
+      end
+
+      def push_comments_to_svn
+        if @app_settings['github'] 
+          push_comments_to_github 
+        elsif @app_settings['gitlab'] 
+          push_comments_to_gitlab
+        end
+      end
+
+      def push_comments_to_github
+        user_name = @app_settings['github_user_name']
+        repo_name = @app_settings['github_repo_name']
+        api_key = @app_settings['github_api_key']
+        merge_request_id = Config.merge_request_id
+        unless [user_name, repo_name, api_key, merge_request_id].all?(&:nil?)
+          HTTParty.post("https://api.github.com/repos/#{user_name}/#{repo_name}/issues/#{merge_request_id}/comments",
+          :body => {'body' => build_note}.to_json,
+          :headers => { 'Authorization' => "token #{api_key}", 'User-Agent' => "#{repo_name}" })
+        end
+      end
+
+      def push_comments_to_gitlab
+        app_id = @app_settings['gitlab_app_id']
+        secret = @app_settings['gitlab_secret']
+        gitlab_url = @app_settings['gitlab_url']
+        merge_request_id = Config.merge_request_id
+        unless [app_id, secret, gitlab_url, merge_request_id].all?(&:nil?)
+          HTTParty.post("#{gitlab_url}/api/v3/projects/#{app_id}/merge_requests/#{merge_request_id}/notes",
+          :query => {'body' => build_note},
+          :headers => {'Private-Token' => secret} )
+        end
+      end
+
+      def build_note  
+        ERB.new(File.read(File.join(File.dirname(__FILE__), '../generators/html/templates/note.html.erb')), nil, '-').result(binding).delete("\n") + "\n"
+      end
+
+      def compare_code_quality
+        Config.base_branch_score > Config.feature_branch_score ? Config.quality_flag = false : Config.quality_flag = true
+        status_reporter.status_message = Config.quality_flag ? "GOOOOOOOOD" : "BAAAAAAAD #{@files_affected} files degraded."
```

Comment:
Was this `Good` / `Bad` present in previous code? If not remove, we don't want to change stdout unless absolutely required.